src/configuration.jl

```diff
@@ -94,22 +94,27 @@ By default, it will be set to 0.0 if there is only one integrand (e.g., length(d
 - `userdata`: User data you want to pass to the integrand and the measurement
 """
 function Configuration(;
-    var::Union{Variable,AbstractVector,Tuple}=(Continuous(0.0, 1.0),),
-    dof::Union{Int,AbstractVector,AbstractMatrix}=((var isa Variable) ? 1 : [ones(Int, length(var)),]),
+    var::V=(Continuous(0.0, 1.0),),
+    dof::Union{Int,AbstractVector,AbstractMatrix}=(is_variable(typeof(var)) ? 1 : [ones(Int, length(var)),]),
     type=Float64,  # type of the integrand
     obs::AbstractVector=zeros(type, length(dof)),
     reweight::Vector{Float64}=ones(length(dof) + 1),
     seed::Int=rand(Random.RandomDevice(), 1:1000000),
     neighbor::Union{Vector{Vector{Int}},Vector{Tuple{Int,Int}},Nothing}=nothing,
     userdata=nothing,
     kwargs...
-)
-    if var isa Variable
+) where {V}
+    if is_variable(V)
         var = (var,)
-    elseif var isa AbstractVector
-        var = (v for v in var)
+    elseif (V <: AbstractVector) || (V <: Tuple)
+        @assert all(v -> is_variable(typeof(v)), var) "All elements in var should be derived from the abstract type Variable, now got $(is_variable.(var))"
+        if var isa AbstractVector
+            var = (v for v in var)
```

Comment:
Maybe `var = collect(var)` works here?

Reply:
we want var to be a tuple. collect returns a vector